### PR TITLE
All roles can join squads

### PR DESCRIPTION
**Pull request by breadhunt**
## About The Pull Request

Lets all roles (from tadpole officer to vehicle crew to researcher to captain) join squads. 

## Why It's Good For The Game

Intended to be merged alongside opt-in squad radio TTS. (will finish the PR in a bit)
After working with an escort squad doing researcher digsites/tadpole crew together, it does get rather hard to communicate with a set group of people when they're offscreen, owing to the fact that _a singular "Common" channel is fucking hell to get anybody's attention in, ever._ Which is a shame because they're all fun with a teamwork focus.

Squads provide a few ways to counter this, namely 1) squad announcements, 2) a radio colour that sets each other apart and 3) a visual indicator on the minimap of who's in your squad and not. This is a minor change but (I think) a good one nonetheless.

..also why the fuck did we have this in the first place? I'm pretty sure the only reason we had it was a holdover from CM's decade-old overwatch shitcode.

## Changelog

:cl:
add: Squads are no longer restricted to marine roles (squaddies, medics, SGs, SLs)
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
Adding squads to shipside seems kinda wacky, considering that if anybody wants to gain access to squad radio, they can take it in cargo and screw the frequency key out of it.

**Comment by breadhunt** [comment]
> Adding squads to shipside seems kinda wacky, considering that if anybody wants to gain access to squad radio, they can take it in cargo and screw the frequency key out of it.

Nope. Frequency keys don't fit in all radios.

Being part of a squad also means you receive/can send out squad messages

**Comment by DrAmazing343** [comment]
Is there a reason we don't allow radios to be manually adjusted, frequency-wise?

**Comment by breadhunt** [comment]
> Is there a reason we don't allow radios to be manually adjusted, frequency-wise?

Probably listening on on HvH frequencies

**Comment by XElectricX** [comment]
Go a step further and allow officers to make squads. CSE-led engi squad, CMO-led medical squad, RO making a cargo squad for their req point farmers... endless funny possibilities.

File: code/datums/jobs/squads.dm [hunk 1]
```diff
@@ -187,10 +187,9 @@
 	if(new_squaddie.assigned_squad)
 		CRASH("attempted to insert marine [new_squaddie] into squad while already having one")
 
-	if(!(new_squaddie.job.title in current_positions))
-		CRASH("Attempted to insert [new_squaddie.job.title] into squad [name]")
-
-	current_positions[new_squaddie.job.title]++
```
Comment (QualityVan) [inline code/datums/jobs/squads.dm:193]: Deleting this means if someone joins in a job already listed they won't increase current positions, the line below only handles unlisted ones.

**Comment by jroinc** [comment]
> Is there a reason we don't allow radios to be manually adjusted, frequency-wise?

Ancient history on CM; people were listening in on the pred frequencies manually.

**Pull request closed without merging**